File: catalog/src/main/java/com/apapedia/catalog/model/Catalog.java
package com.apapedia.catalog.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.UUID;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "catalog")
// @JsonIgnoreProperties(value={"category"}, allowSetters = true)
public class Catalog {
    @Id
    @NotNull
    private UUID id = UUID.randomUUID();

    @Column(name = "seller_id", nullable = true)
    private UUID sellerId;

    @Column(name = "price", nullable = false)
    private int price;

    @Column(name = "product_name", nullable = false)
    private String productName;

    @Column(name = "product_description", nullable = false)
    private String productDescription;

    @JsonIgnore
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "category_id", referencedColumnName = "categoryId")
    private Category category;

    @Column(name = "stock", nullable = false)
    private int stock;

    @Column(name = "image", nullable = true)
    private String image;

    @Column(name = "is_deleted", nullable = false)
    private boolean isDeleted = false;

}

File: apapedia_mobile/lib/domain/Catalog.dart


class Catalog {
  final String id;
  final String sellerId;
  final int price;
  final String productName;
  final String productDescription;
  final int stock;
  final List<int> image;
  final bool isDeleted;

  Catalog({
    required this.id,
    required this.sellerId,
    required this.price,
    required this.productName,
    required this.productDescription,
    required this.stock,
    required this.image,
    required this.isDeleted,
  });

  factory Catalog.fromJson(Map<String, dynamic> json) => Catalog(
        id: json['id'],
        sellerId: json['sellerId'],
        price: json['price'],
        productName: json['productName'],
        productDescription: json['productDescription'],
        stock: json['stock'],
        image: List<int>.from(json['image']),
        isDeleted: json['isDeleted'],
      );

  Map<String, dynamic> toJson() => {
        'id': id,
        'sellerId': sellerId,
        'price': price,
        'productName': productName,
        'productDescription': productDescription,
        'stock': stock,
        'image': image,
        'isDeleted': isDeleted,
      };
}